auslib/test/blobs/test_base.py

```diff
@@ -63,3 +67,132 @@ def testSchemaCaching(self):
             blob.validate('fake', [])
 
             self.assertEquals(yaml_load.call_count, 1)
+
+
+# Things we consider to be useful values in testing blobs. Basically, these
+# are things would actually show up in real blobs. Nesting is handled
+# further down.
+# Ideally we'd include st.boolean() here, but without a way to exclude
+# booleans from list items (which doesn't happen in the real world), we end
+# up with test failures because things like "1 in [True]" are True, which
+# ends up failing tests.
+useful_values = st.none() | st.floats(allow_nan=False) | st.text()
+
+
+@given(st.lists(useful_values), st.lists(useful_values), st.lists(useful_values))
+def test_merge_lists_unique_items_present(base, left_additional, right_additional):
+    left = deepcopy(base) + left_additional
+    right = deepcopy(base) + right_additional
+    expected = sorted(list(set(deepcopy(base) + left_additional + right_additional)))
+    got = sorted(merge_lists(base, left, right))
+    assert got == expected
+
+
+@given(st.lists(useful_values), st.lists(useful_values))
+def test_merge_lists_no_dupes(base, additional):
+    left = deepcopy(base) + additional
+    right = deepcopy(base) + additional
+    expected = sorted(list(set(deepcopy(base) + additional)))
+    got = sorted(merge_lists(base, left, right))
```

Comment:
``expected = left; got = merge_lists(base, left, right)`` or is there some subtlety I'm missing ?

Reply:
Unfortunately, we need to de-dupe `expected` because we may get tests like:
`Falsifying example: test_merge_lists_no_dupes(base=[None, None], additional=[])`

And once we're de-duping, casting back to a sorted list is required to avoid failures due to ordering differences.

This can be simplified to use `left` as the base for expected, but not any further as far as I can tell.

Reply:
OK, so ``base`` may have dupes and merge_lists will remove them, and the test needs to account for that. Does that reflect a useful situation for Balrog ? Would it make sense to dedupe base ?